app/src/main/java/com/sdp13epfl2021/projmag/activities/ProfilePageActivity.kt

```diff
@@ -30,23 +43,73 @@ class ProfilePageActivity : AppCompatActivity() {
             val gallery = Intent(Intent.ACTION_PICK, MediaStore.Images.Media.INTERNAL_CONTENT_URI)
             startActivityForResult(gallery, pickImage)
         }
+
+
+        UserProfileDatabase(Firebase.firestore, Firebase.auth).getProfile(::loadUserProfile)
+
         buttonAddCv.setOnClickListener{
             val intent = Intent(this,CVCreationActivity::class.java)
             startActivity(intent)
             finish()
         }
         buttonSubChange.setOnClickListener{
+            val profile = createProfileFromFields()
+            if(profile != null) {
+                UserProfileDatabase(Firebase.firestore, Firebase.auth).uploadProfile(profile, {}, {})
+            }
             val intent = Intent(this,ProjectsListActivity::class.java)
             startActivity(intent)
             finish()
         }
+
+    }
+
+    private fun loadUserProfile(profile : ImmutableProfile?) {
+        if (profile != null) {
+            Log.d(ContentValues.TAG, "NOT NULL")
+            findViewById<EditText>(R.id.profile_firstname).setText(profile.firstName)
+            findViewById<EditText>(R.id.profile_lastname).setText(profile.lastName)
+            findViewById<EditText>(R.id.profile_age).setText(profile.age.toString())
+            findViewById<EditText>(R.id.profile_genre).setText(profile.gender.toString())
+            findViewById<EditText>(R.id.profile_phone_number).setText(profile.phoneNumber)
+            findViewById<EditText>(R.id.profile_sciper).setText(profile.sciper.toString())
+        } else {
+            Log.d(ContentValues.TAG, "NULL")
+        }
     }
 
+
     override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
         super.onActivityResult(requestCode, resultCode, data)
         if (resultCode == RESULT_OK && requestCode == pickImage) {
             imageUri = data?.data
             imageView.setImageURI(imageUri)
         }
     }
+
+    private fun createProfileFromFields() : ImmutableProfile? {
+        val firstName = findViewById<EditText>(R.id.profile_firstname).text.toString()
+        val lastName = findViewById<EditText>(R.id.profile_lastname).text.toString()
+        val age = findViewById<EditText>(R.id.profile_age).text.toString()
+        val gender = if (findViewById<EditText>(R.id.profile_genre).text.toString() == Gender.MALE.toString())  Gender.MALE
+        else {
+            if (findViewById<EditText>(R.id.profile_genre).text.toString() == Gender.FEMALE.toString()) Gender.FEMALE else Gender.OTHER
+        }
```

Comment:
There is some code dupplication, maybe using a when (...) {...} would be more readable.

Reply:
I agree I can do that next sprint because I wont have the time to modify the code and the tests before tomorrows sprint